src/test/java/com/github/cameltooling/lsp/internal/completion/modeline/CamelKModelineInsertionTest.java

```diff
@@ -0,0 +1,368 @@
+package com.github.cameltooling.lsp.internal.completion.modeline;
+
+import com.github.cameltooling.lsp.internal.AbstractCamelLanguageServerTest;
+import com.github.cameltooling.lsp.internal.CamelLanguageServer;
+import org.eclipse.lsp4j.CompletionItem;
+import org.eclipse.lsp4j.CompletionList;
+import org.eclipse.lsp4j.Position;
+import org.eclipse.lsp4j.jsonrpc.messages.Either;
+import org.junit.jupiter.api.Test;
+
+import java.util.List;
+import java.util.concurrent.CompletableFuture;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+/**
+ @author joshiraez
+ */
+class CamelKModelineInsertionTest extends AbstractCamelLanguageServerTest {
+
+    /** EMPTY FILE TESTS **/
+    @Test
+    void testProvideInsertionOnEmptyXMLFile() throws Exception {
+        FileType type = FileType.XML;
+        String contents = "";
+        Position position = beginningOfLastLine(contents);
+
+        List<CompletionItem> completionItems = getCompletionsFor(type, contents, position);
+
+        assertCompletionItemsHasExpectedCompletionForType(type, completionItems);
+    }
+
+    @Test
+    void testProvideInsertionOnEmptyJavaFile() throws Exception {
+        FileType type = FileType.Java;
+        String contents = "";
+        Position position = beginningOfLastLine(contents);
+
+        List<CompletionItem> completionItems = getCompletionsFor(type, contents, position);
+
+        assertCompletionItemsHasExpectedCompletionForType(type, completionItems);
+    }
+
+    @Test
+    void testProvideInsertionOnEmptyYAMLFile() throws Exception {
+        FileType type = FileType.YAML;
+        String contents = "";
+        Position position = beginningOfLastLine(contents);
+
+        List<CompletionItem> completionItems = getCompletionsFor(type, contents, position);
+
+        assertCompletionItemsHasExpectedCompletionForType(type, completionItems);
+    }
+
+    @Test
+    void testProvideInsertionOnEmptyYMLFile() throws Exception {
+        String extension = ".camelk.yml";
+        String contents = "";
+        Position position = beginningOfLastLine(contents);
+
+        CamelLanguageServer camelLanguageServer = initializeLanguageServer(contents, extension);
+        CompletableFuture<Either<List<CompletionItem>, CompletionList>> completions = getCompletionFor(camelLanguageServer, position);
+        List<CompletionItem> completionItems =  completions.get().getLeft();
+
+        assertThat(completionItems).hasSize(1);
+        assertThat(completionItems).contains(FileType.YAML.completion);
```

Comment:
chaning the Assertj statements allows to have a more precise error message in case of failure.

```
assertThat(completionItems)
.hasSize(1);
.contains(FileType.YAML.completion);
```